receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -83,27 +82,24 @@ func (s *scraper) start(context.Context, component.Host) error {
 }
 
 func (s *scraper) scrape(_ context.Context) (pdata.Metrics, error) {
-	md := pdata.NewMetrics()
-	rms := md.ResourceMetrics()
 
 	var errs scrapererror.ScrapeErrors
 
-	metadata, err := s.getProcessMetadata()
+	mdata, err := s.getProcessMetadata()
 	if err != nil {
 		partialErr, isPartial := err.(scrapererror.PartialScrapeError)
 		if !isPartial {
-			return md, err
+			return pdata.NewMetrics(), err
 		}
 
 		errs.AddPartial(partialErr.Failed, partialErr)
 	}
 
-	rms.EnsureCapacity(len(metadata))
-	for _, md := range metadata {
-		rm := rms.AppendEmpty()
-		rm.SetSchemaUrl(conventions.SchemaURL)
-		md.initializeResource(rm.Resource())
-		metrics := rm.InstrumentationLibraryMetrics().AppendEmpty().Metrics()
+	resource := pdata.NewResource()
+
+	for _, md := range mdata {
+
+		md.initializeResource(resource)
```

Comment:
One solution would be to create another builder at the resource level, and move the Record... functions to that. 

The scraper author can ask the MetricsBuilder for a new ResourceBuilder as needed, attach attributes and data points to it, and then ultimately call `Emit` on the MetricsBuilder. Since the MetricsBuilder created the ResourceBuilders, it can call `emit` on each of those in turn.

Reply:
`Emit` can be called with an option `WithResource` which copies the resource into the generate metrics, would that not work here?

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8382/files#diff-806ca51937857b079bdde5bf7e3e87aeb773245b2cb93b40078d241515bdab68R138-R142

Reply:
Ah i see.. Yes the code is limited to a single resource for each call to Emit which means the only way to support multiple resources at this time is with separate calls to emit. Let me see what this looks like with a resource builder

Reply:
I'm thinking a bit different approach to this problem that also solves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7301. Let me submit an alternative PR and we can continue the discussion then